components/sync/driver/sync_service_impl_delegate.h

```diff
@@ -18,12 +18,11 @@ class SyncServiceImplDelegate {
   virtual void SuspendDeviceObserverForOwnReset() = 0;
   virtual void ResumeDeviceObserver() = 0;
 
-  void set_profile_sync_service(BraveSyncServiceImpl* sync_service_impl) {
-    sync_service_impl_ = sync_service_impl;
-  }
+  virtual void SetLocalDeviceAppearedCallback(
+      base::OnceCallback<void()> local_device_appeared_callback) = 0;
 
- protected:
-  raw_ptr<BraveSyncServiceImpl> sync_service_impl_ = nullptr;
+  virtual void set_profile_sync_service(
+      BraveSyncServiceImpl* sync_service_impl) = 0;
```

Comment:
why are we changing this? `sync_service_impl_` is already accessible in `BraveSyncServiceImplDelegate`

Reply:
this is mistake, fixed at https://github.com/brave/brave-core/pull/13562/commits/d82d0710faaf621d336396bf36e8a8ff779b11ed

thanks for pointing this